lib/types/cql-utils.js

```diff
@@ -139,6 +139,46 @@ function encodeTuple(value, type) {
     return [rust.ComplexType.remapTupleSupportType(newTypes), res];
 }
 
+/**
+ * Wraps UDT into format recognized by ParameterWrapper
+ * @param {Object} value
+ * @param {rust.ComplexType} type
+ * @returns {Array<rust.ComplexType|Array<Array<String|any>>>} Returns tuple: [rust.ComplexType, [[fieldName, fieldValue]]]
+ */
+function encodeUdt(value, type) {
+    let names = type.getUdtFieldNames();
+    let types = type.getInnerTypes();
+    let newTypes = [];
+    let res = [];
+
+    if (names.length !== types.length) {
+        throw new TypeError(
+            `Expected ${types.length} field names in UDT, obtained ${names.length}`,
+        );
+    }
+
+    for (let i = 0; i < names.length; i++) {
+        if (!types[i]) {
+            types[i] = guessTypeChecked(value[names[i]]);
+        }
+        const element = getWrapped(
+            types[i],
+            value[names[i]] !== undefined ? value[names[i]] : null, // Undefineds should be treated as nulls, so we convert them to nulls
+        );
+        newTypes.push(element[0]);
+        res.push([names[i], element[1]]);
+    }
+    return [
+        rust.ComplexType.remapUdtSupportType(
+            newTypes,
+            names,
+            type.getUdtKeyspace(),
+            type.getUdtName(),
+        ),
+        res,
+    ];
```

Comment:
:wrench: The name `remapUdtSupportType` is extremely confusing. Please make it better.